dashboard/modules/job/tests/test_cli_integration.py

```diff
@@ -19,12 +19,13 @@ def set_env_var(key: str, val: Optional[str] = None):
     elif key in os.environ:
         del os.environ[key]
 
-    yield
-
-    if key in os.environ:
-        del os.environ[key]
-    if old_val is not None:
-        os.environ[key] = old_val
+    try:
```

Comment:
what is this for btw?

Reply:
This makes the tests less leaky in the case of an exception being thrown in the test code. If there is an exception during the test, previously, the env vars were not being reset.